array-filter/main.js

```diff
@@ -0,0 +1,21 @@
+const numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const names = [
+  'Ada',
+  'Hedy',
+  'Jean',
+  'Grace',
+  'Evelyn',
+  'Joan',
+  'Elizabeth',
+  'Janese',
+  'Donna'
+];
+
+const evenNumbers = numbers.filter((number) => !(number % 2));
+const overFive = numbers.filter((number) => number > 5);
+// not sure which is best.
+// const startWithE = names.filter((name) => name[0] === 'E');
+// const startWithE = names.filter((name) => !name.indexOf('E'));
```

Comment:
Give those two choices (line 18 and line 19), I would choose line 18 (`name[0] === 'E'`) because when I read line 19, I have to stop and work through the logic to convince myself it's correct. Also, the fact that the number 0 is falsy is a rather unfortunate aspect of JavaScript and I would hate to see code actually _relying_ on that! ;-)